### PR TITLE
DOC: added apply example to core/dataarray.py

### PR DESCRIPTION
**Pull request by raybellwaves**
- [NA] Closes #xxxx
 - [NA] Tests added / passed
 - [ ] Passes ``git diff upstream/master | flake8 --diff``
 - [NA] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

I added an example to apply as I just learnt about it from my Q on stackoverflow (https://stackoverflow.com/questions/46980823/python-xarray-compute-the-absolute-values-of-a-dataset/46981027#46981027).

Also my first time using git and making edits on the command line rather than the web browser.

### CODE REVIEW

File: xarray/core/dataset.py [hunk 1]
```diff
@@ -2431,6 +2431,25 @@ def apply(self, func, keep_attrs=False, args=(), **kwargs):
         attrs = self.attrs if keep_attrs else None
         return type(self)(variables, attrs=attrs)
 
+        Examples
```
Comment (shoyer) [inline xarray/core/dataset.py]: This needs to be in the docstring, not the python code!

File: xarray/core/dataset.py [hunk 2]
```diff
@@ -2431,6 +2431,25 @@ def apply(self, func, keep_attrs=False, args=(), **kwargs):
...
+        --------
```
Comment (shoyer) [inline xarray/core/dataset.py]: add an extra line break here

**Comment by shoyer** [comment]
Thanks @raybellwaves! This will be helpful.

File: xarray/core/dataset.py (hunk 1 above)
Comment (raybellwaves) [reply xarray/core/dataset.py]: I shouldn't do PR's on a Friday afternoon!

**Comment by shoyer** [comment]
Thanks!

**Pull request merged**